Samples/Weather.cs

```diff
@@ -19,28 +19,118 @@
 using System.Text;
 using System.Text.Json;
+using System.Runtime.InteropServices;
 
 namespace OpenZiti.Samples {
+
     public class Weather {
         static MemoryStream ms = new MemoryStream(2 << 16); //a big bucket to hold bytes to display contiguously at the end of the program
+        static ZitiTunnelCommand.Options tunOptions = new ZitiTunnelCommand.Options();
+
+        static ZitiInstance zitiInstance = new ZitiInstance();
+
+        internal static void OnZitiTunnelNextAction(object sender, ZitiTunnelCommand.NextAction action)
+		{
+            string mfacode;
+
+            switch (action.command)
+			{
+                case 1:
+					{
+                        Console.WriteLine("Enable MFA for the identity: " + (zitiInstance.Zid?.IdentityNameFromController != null ? zitiInstance.Zid?.IdentityNameFromController : zitiInstance.Zid?.InitOpts.IdentityFile)); // list the identities and allow user to choose
+                        ZitiIdentity.TunnelCB tunnelCB = new ZitiIdentity.TunnelCB();
+                        tunnelCB.zidOpts = zitiInstance.Zid.InitOpts;
+                        ZitiIdentity.TunnelCB.ZitiResponseDelegate cbDelegate = tunnelCB.ZitiResponse;
+                        ZitiMFAService.ziti_mfa_enroll(zitiInstance.Zid, Marshal.GetFunctionPointerForDelegate(cbDelegate));
+                        break;
+                    }
+                case 2:
+					{
+                        Console.WriteLine("Verify MFA for the identity" + (zitiInstance.Zid?.IdentityNameFromController != null ? zitiInstance.Zid?.IdentityNameFromController : zitiInstance.Zid?.InitOpts.IdentityFile));
+                        Console.WriteLine("Enter the mfa auth codo: ");
+                        mfacode = Console.ReadLine();
+                        ZitiIdentity.TunnelCB tunnelCB = new ZitiIdentity.TunnelCB();
+                        tunnelCB.zidOpts = zitiInstance.Zid.InitOpts;
+                        ZitiIdentity.TunnelCB.ZitiResponseDelegate cbDelegate = tunnelCB.ZitiResponse;
+                        ZitiMFAService.verify_mfa(zitiInstance.Zid, mfacode, Marshal.GetFunctionPointerForDelegate(cbDelegate));
+                        break;
+                    }
+                case 3:
+                    {
+                        Console.WriteLine("Remove MFA for the identity" + (zitiInstance.Zid?.IdentityNameFromController != null ? zitiInstance.Zid?.IdentityNameFromController : zitiInstance.Zid?.InitOpts.IdentityFile));
+                        Console.WriteLine("Enter the mfa auth codo: ");
+                        mfacode = Console.ReadLine();
+                        ZitiIdentity.TunnelCB tunnelCB = new ZitiIdentity.TunnelCB();
+                        tunnelCB.zidOpts = zitiInstance.Zid.InitOpts;
+                        ZitiIdentity.TunnelCB.ZitiResponseDelegate cbDelegate = tunnelCB.ZitiResponse;
+                        ZitiMFAService.remove_mfa(zitiInstance.Zid, mfacode, Marshal.GetFunctionPointerForDelegate(cbDelegate));

```

Comment:
i'll make the same comments below probably but "ZitiMFAService" is a bad name and probably all these functions just belong on API.cs